internal/passwordgenerator/password_generator.go

```diff
@@ -0,0 +1,60 @@
+package passwordgenerator
+
+import (
+	crypto "crypto/rand"
+	"math/big"
+	"strings"
+)
+
+const (
+	numbers        = "0123456789"
+	lowerLetters   = "abcdedfghijklmnopqrstuvwxyz"
+	upperLetters   = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+	specialSymbols = "!$%^&*()_+{}:@[];'#<>?,./|\\\\-=?"
+	allSet         = lowerLetters + upperLetters + specialSymbols + numbers
+)
+
+func GeneratePassword(length, minNumbers, minLower, minUpper, minSymbol int) string {
+	var password strings.Builder
+
+	for i := 0; i < minNumbers; i++ {
+		random := randInt(len(numbers))
+		password.WriteString(string(numbers[random]))
+	}
+
+	for i := 0; i < minUpper; i++ {
+		random := randInt(len(upperLetters))
+		password.WriteString(string(upperLetters[random]))
+	}
+
+	for i := 0; i < minLower; i++ {
+		random := randInt(len(lowerLetters))
+		password.WriteString(string(lowerLetters[random]))
+	}
+
+	for i := 0; i < minSymbol; i++ {
+		random := randInt(len(specialSymbols))
+		password.WriteString(string(specialSymbols[random]))
+	}
+
+	remainingLength := length - minNumbers - minLower - minUpper - minSymbol
+	for i := 0; i < remainingLength; i++ {
+		random := randInt(len(allSet))
+		password.WriteString(string(allSet[random]))
+	}
+	inRune := []rune(password.String())
+	for i := range inRune {
+		j := randInt(i + 1)
+		inRune[i], inRune[j] = inRune[j], inRune[i]
+	}
+
+	return string(inRune)
```

Comment:
Are all Numbers, upper and lower always returned in the same order?

Reply:
Yes this part does the shuffle: 
```
inRune := []rune(password.String())
	for i := range inRune {
		j := randInt(i + 1)
		inRune[i], inRune[j] = inRune[j], inRune[i]
	}
```